src/native.rs

```diff
@@ -2123,6 +2124,74 @@ impl HasContext for Context {
         gl.GetQueryObjectuiv(query, parameter, &mut value);
         value
     }
+
+    unsafe fn create_transform_feedback(&self) -> Result<Self::TransformFeedback, String> {
+        let gl = &self.raw;
+        let mut name = 0;
+        gl.GenTransformFeedbacks(1, &mut name);
+        Ok(name)
+    }
+
+    unsafe fn delete_transform_feedback(&self, transform_feedback: Self::TransformFeedback) {
+        let gl = &self.raw;
+        gl.DeleteTransformFeedbacks(1, &transform_feedback);
+    }
+
+    unsafe fn bind_transform_feedback(&self, target: u32, transform_feedback: Option<Self::TransformFeedback>) {
+        let gl = &self.raw;
+        gl.BindTransformFeedback(target, transform_feedback.unwrap_or(0));
+    }
+
+    unsafe fn begin_transform_feedback(&self, primitive_mode: u32) {
+        let gl = &self.raw;
+        gl.BeginTransformFeedback(primitive_mode);
+    }
+
+    unsafe fn end_transform_feedback(&self) {
+        let gl = &self.raw;
+        gl.EndTransformFeedback();
+    }
+
+    unsafe fn pause_transform_feedback(&self) {
+        let gl = &self.raw;
+        gl.PauseTransformFeedback();
+    }
+
+    unsafe fn resume_transform_feedback(&self) {
+        let gl = &self.raw;
+        gl.ResumeTransformFeedback();
+    }
+
+    unsafe fn transform_feedback_varyings(&self, program: Self::Program, varyings: &[&str], buffer_mode: u32) {
+        let gl = &self.raw;
+        gl.TransformFeedbackVaryings(
+            program,
+            varyings.len() as i32,
+            varyings.as_ptr() as *const *const native_gl::types::GLchar,
+            buffer_mode,
+        );
+    }
+
+    unsafe fn get_transform_feedback_varying(&self, program: Self::Program, index: u32) -> Option<(i32, u32, String)> {
+        let gl = &self.raw;
+
+        let buf_size: i32 = 256;
+        let size: i32 = 0;
+        let ty: u32 = 0;
+        let mut name = String::with_capacity(buf_size as usize);
+
+        gl.GetTransformFeedbackVarying(
+            program,
+            index,
+            buf_size,
+            std::ptr::null_mut(),
+            size as *mut i32,
+            ty as *mut u32,
+            name.as_mut_ptr() as *mut i8
+        );
```

Comment:
And also this!

Reply:
We'll need to do the opposite here, the `CString` documentation talks about this a bit, but the idea is that we could copy something like `CString::from_raw` into our own `String`